packages/core/src/classes.ts

```diff
@@ -59,35 +59,56 @@ export type AllAttributes<T extends E, E extends Obj> = Required<Attributes<T, E
 
 export enum ClassifierKind {
   CLASS,
-  MIXIN
+  MIXIN,
+  ENUM
 }
 
-export interface EClassifier<T extends E, E extends Obj> extends Doc {
+export interface Classifier extends Doc {
   _kind: ClassifierKind
-  _attributes: AllAttributes<T, E>
-  _extends?: Ref<Classifier<E>>
 }
 
-export type Classifier<T extends Obj> = EClassifier<T, Obj>
-
 export interface EMixin<T extends E, E extends Obj> extends EClass<T, E> {
 }
 
 export type Mixin<T extends Obj> = EMixin<T, Obj>
 
-export interface EClass<T extends E, E extends Obj> extends EClassifier<T, E> {
-  _native?: StringProperty
+export interface EDomainClassifier {
   _domain?: StringProperty
 }
 
+export interface EClass<T extends E, E extends Obj> extends Classifier, EDomainClassifier {
+  _attributes: AllAttributes<T, E>
+  _extends?: Ref<Class<E>>
+
+  _native?: StringProperty
+}
+
 export type Class<T extends Obj> = EClass<T, Obj>
 
+export interface EnumLiteral extends Emb {
+  label: string
+  ordinal: string | number
+}
+
+export type EnumKey = string | number | symbol
+export type EnumLiterals<T extends EnumKey, E extends EnumLiteral> = { [Q in T]: E }
+
+export interface EEnum<T extends EnumKey, E extends EnumLiteral> extends Classifier, EDomainClassifier {
+  _literals: EnumLiterals<T, E>
+}
+
+export type Enum<T extends EnumKey> = EEnum<T, EnumLiteral>
```

Comment:
I just started to think why `Enum` is derived from `Classifier` and not from a `Type` :)